src/Knp/Rad/Security/Voter/IsOwnerVoter.php

```diff
@@ -2,49 +2,54 @@
 
 namespace Knp\Rad\Security\Voter;
 
-use Symfony\Component\Security\Core\Authorization\Voter\VoterInterface;
+use Knp\Rad\Security\OwnableInterface;
+use Knp\Rad\Security\OwnerInterface;
 use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\VoterInterface;
 use Symfony\Component\Security\Core\User\EquatableInterface;
 use Symfony\Component\Security\Core\User\UserInterface;
-use Knp\Rad\Security\OwnerInterface;
-use Knp\Rad\Security\OwnableInterface;
 
 class IsOwnerVoter implements VoterInterface
 {
     const IS_OWNER = 'IS_OWNER';
 
+    /**
+     * {@inheritdoc}
+     */
     public function supportsAttribute($attribute)
     {
         return self::IS_OWNER === $attribute;
     }
 
+    /**
+     * {@inheritdoc}
+     */
     public function supportsClass($class)
     {
-        if (is_object($class)) {
-            $refl = new \ReflectionObject($class);
-
-            return $refl->implementsInterface('Knp\Rad\Security\OwnableInterface');
-        }
-
-        return false;
+        return is_subclass_of($class, 'Knp\Rad\Security\OwnableInterface');
     }
 
+    /**
+     * {@inheritdoc}
+     */
     public function vote(TokenInterface $token, $object, array $attributes)
     {
+        $class = true === is_object($object) ? get_class($object) : $object;
```

Comment:
We should not try to vote for null `$object` at all. Also `$object` will be always an object.